x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -724,6 +906,87 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     }
   }
 
+  /** Unlike `ensureAuthorized`, this accepts multiple actions, and it does not utilize legacy audit logging */
+  private async ensureAuthorizedV2<T extends string>(
```

Comment:
> If this is the future direction for Saved Objects authorization (the intent isn't really clear to me), then I'd prefer to call this `ensureAuthorized`, and update the existing function to `legacyEnsureAuthorized`.

I like the that better, I can certainly rename these accordingly.

> I'd also like to see if there's a way to consolidate the authorization logic so that we don't have to maintain two complex functions that are largely responsible for the same thing.

I struggled with this, I wanted to be able to make a single privilege check and handle the actual authorization decision inside the API functions. At first I modified `ensureAuthorized` to accept additional options (to allow multiple actions, to skip legacy audit logging, etc), but I quickly realized the complexity made it even more inscrutable than it already is. Then I tried to consolidate some of this logic so it could be reused between the two functions but it just came out messy too.

I thought this was a better alternative, and yeah, it may be a better future direction for Saved Objects authorization so we can cut down on our ES privilege checks.